infra/modules/curi/api_gateway/main.tf

```diff
@@ -1,49 +1,45 @@
-resource "aws_apigatewayv2_api" "example" {
-  name          = "${terraform.workspace}-example"
+resource "aws_apigatewayv2_api" "lambda_gw" {
+  name          = "${terraform.workspace}-lambda-gw"
   protocol_type = "HTTP"
 }
 
-# resource "aws_apigatewayv2_authorizer" "example" {
-#   api_id           = aws_apigatewayv2_api.example.id
-#   authorizer_type  = "JWT"
-#   identity_sources = ["$request.header.Authorization"]
-#   name             = "example-authorizer"
+resource "aws_cognito_user_pool" "lambda_gw_pool" {
+  name = "${terraform.workspace}-lambda-gw-pool"
+}
+
+resource "aws_cognito_user_pool_client" "lambda_gw_pool_client" {
+  name = "${terraform.workspace}-lambda-gw-pool-client"
+
+  user_pool_id = aws_cognito_user_pool.lambda_gw_pool.id
+
+  # allowed_oauth_flows_user_pool_client = true
+  # allowed_oauth_flows - (Optional) List of allowed OAuth flows (code, implicit, client_credentials).
+  # allowed_oauth_scopes - (Optional) List of allowed OAuth scopes (phone, email, openid, profile, and aws.cognito.signin.user.admin).
+  explicit_auth_flows = ["USER_PASSWORD_AUTH"]
+}
 
-#   jwt_configuration {
-#     audience = ["example"]
-#     issuer   = "https://${aws_cognito_user_pool.example.endpoint}"
-#   }
-# }
+resource "aws_apigatewayv2_authorizer" "lambda_gw_auth" {
+  api_id           = aws_apigatewayv2_api.lambda_gw.id
+  authorizer_type  = "JWT"
+  identity_sources = ["$request.header.Authorization"]
+  name             = "${terraform.workspace}-lambda-gw-authorizer"
+
+  jwt_configuration {
+    audience = [aws_apigatewayv2_stage.lambda_gw_stage.invoke_url]
```

Comment:
This is the problem,  I think the audience should be the client_id for the cognito user pool